### ISSUE TITLE
Using config globals with quotes throws error, but using globals without quotes still continues.

### ISSUE THREAD

**Issue by Zenwolf** [eslint/eslint#358]
In the `.eslintrc` config JSON, using globals with quotes (valid JSON) fails:

``` json
{
    "globals": {
        "foo": false
    }
}
```

But this invalid JSON without quotes doesn't throw the error (although the globals still aren't acknowledged):

``` json
{
    globals: {
        "foo": false
    }
}
```

Error:

``` text
/Users/somebody/git/eslint/lib/util.js:33
            base[key] = custom[key];
                      ^
TypeError: Cannot set property 'foo' of undefined
    at /Users/somebody/git/eslint/lib/util.js:33:23
    at Array.forEach (native)
    at mergeConfigs (/Users/somebody/git/eslint/lib/util.js:27:25)
    at /Users/somebody/git/eslint/lib/util.js:31:25
    at Array.forEach (native)
    at Object.mergeConfigs (/Users/somebody/git/eslint/lib/util.js:27:25)
    at Config.mergeConfigs (/Users/somebody/git/eslint/lib/config.js:77:17)
    at Config.getConfig (/Users/somebody/git/eslint/lib/config.js:64:43)
    at processFile (/Users/somebody/git/eslint/lib/cli.js:113:31)
    at /Users/somebody/git/eslint/lib/cli.js:166:27
Task completed in 0.09s
```
